api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListChildrenAdolescentARTWithoutFullDisclosureDataset.java

```diff
@@ -0,0 +1,219 @@
+    sqlPatientDataDefinition.setName("Get ART Start Date");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Date.class));
+
+    sqlPatientDataDefinition.setQuery(commonQueries.getARTStartDate(true));
+
+    return sqlPatientDataDefinition;
+  }
+
+  private DataDefinition getAdolescentsCurrentlyOnArtWithDisclosures() {
+    ObsForPersonDataDefinition obsForPersonDataDefinition = new ObsForPersonDataDefinition();
+    obsForPersonDataDefinition.setName("Adolescent patients with disclosures");
+    obsForPersonDataDefinition.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
+    obsForPersonDataDefinition.addParameter(
+        new Parameter("locationList", "Location", Location.class));
+    obsForPersonDataDefinition.setEncounterTypeList(
+        Arrays.asList(hivMetadata.getPrevencaoPositivaSeguimentoEncounterType()));
+    obsForPersonDataDefinition.setQuestion(
+        hivMetadata.getDisclosureOfHIVDiagnosisToChildrenAdolescentsConcept());
+    obsForPersonDataDefinition.setWhich(TimeQualifier.LAST);
+    obsForPersonDataDefinition.setValueCodedList(
+        Arrays.asList(
+            hivMetadata.getPartiallyRevealedConcept(), hivMetadata.getNotRevealedConcept()));
+
+    return obsForPersonDataDefinition;
+  }
+
+  private DataDefinition getAdolescentsCurrentlyOnArtWithPartialDisclosures(Concept valueCoded) {
+    Map<String, Integer> map = new HashMap<>();
+    map.put("35", hivMetadata.getPrevencaoPositivaSeguimentoEncounterType().getEncounterTypeId());
+    map.put(
+        "6340",
+        hivMetadata.getDisclosureOfHIVDiagnosisToChildrenAdolescentsConcept().getConceptId());
+    map.put("answer", valueCoded.getConceptId());
+    SqlPatientDataDefinition cd = new SqlPatientDataDefinition();
+    cd.setName("Adolescent patients with disclosure of " + valueCoded);
+    cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
+    cd.addParameter(new Parameter("locationList", "Location", Location.class));
+    String query =
+        "SELECT p.patient_id,MIN(e.encounter_datetime) FROM patient p "
+            + " INNER JOIN encounter e ON p.patient_id=e.patient_id "
+            + " INNER JOIN obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND e.encounter_type = ${35} "
+            + " AND o.concept_id=${6340} AND o.value_coded= ${answer} AND e.encounter_datetime <= :onOrBefore "
+            + " AND e.location_id=:locationList "
+            + " GROUP BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    String replacedQuery = sb.replace(query);
+    cd.setQuery(replacedQuery);
+    return cd;
+  }
+
+  private DataDefinition
+      getAdolescentsCurrentlyOnArtWithPartialDisclosuresDifferenceFromReportingDate(
```

Comment:
Spec

Reply:
@ningosi how do you know they Adolescents, is it checked somewhere else?

Reply:
Those are filtered in the base cohort passed. Patients who are between 8 to 14 years see this getBaseCohortForAdolescent() method that combines all